packages/modal-ui/src/lib/components/WalletOptions.tsx

```diff
@@ -76,13 +76,17 @@ export const WalletOptions: React.FC<WalletOptionsProps> = ({
             const { selectedWalletId } = selector.store.getState();
             const { name, description, iconUrl } = module.metadata;
             const selected = module.id === selectedWalletId;
-
             result.push(
               <li
                 key={module.id}
                 id={module.id}
                 className={selected ? "selected-wallet" : ""}
-                onClick={selected ? undefined : handleWalletClick(module)}
+                onClick={
+                  selected
+                    ? // This kinda broke for me if we dont have eager connect
+                      handleWalletClick(module)
```

Comment:
- The way we have it it's meant to avoid calling signIn on a wallet that's already signedIn.
Can you please provide more info what kind of problem you had with this code?

Reply:
This change should be reverted to address the point made by @kujtimprenkuSQA